src/app/clusters/test-cluster-server/test-cluster-server.cpp

```diff
@@ -42,18 +42,27 @@ bool emberAfTestClusterClusterTestCallback(chip::app::Command *)
     return true;
 }
 
-bool emberAfTestClusterClusterTestSpecificCallback(chip::app::Command *)
+bool emberAfTestClusterClusterTestSpecificCallback(chip::app::Command * apCommandObj)
 {
+    CHIP_ERROR err      = CHIP_NO_ERROR;
     uint8_t returnValue = 7;
-    emberAfFillExternalBuffer((ZCL_CLUSTER_SPECIFIC_COMMAND | ZCL_FRAME_CONTROL_SERVER_TO_CLIENT), ZCL_TEST_CLUSTER_ID,
-                              ZCL_TEST_SPECIFIC_RESPONSE_COMMAND_ID, "u", returnValue);
 
-    EmberStatus sendStatus = emberAfSendResponse();
-    if (EMBER_SUCCESS != sendStatus)
+    app::CommandPathParams cmdParams = { emberAfCurrentEndpoint(), /* group id */ 0, ZCL_TEST_CLUSTER_ID,
+                                         ZCL_TEST_SPECIFIC_RESPONSE_COMMAND_ID, (chip::app::CommandPathFlags::kEndpointIdValid) };
+    TLV::TLVWriter * writer          = nullptr;
+
+    VerifyOrExit(apCommandObj != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+
+    SuccessOrExit(err = apCommandObj->PrepareCommand(&cmdParams));
+    writer = apCommandObj->GetCommandDataElementTLVWriter();
+    SuccessOrExit(err = writer->Put(TLV::ContextTag(0), returnValue));
+    SuccessOrExit(err = apCommandObj->FinishCommand());
```

Comment:
I understand this this code will makes happy the Darwin CI. And this is a great example of how to replace `emberAfSendResponse`!
But if I understand correctly, the CI will be green for this particular test command, but in reality all other commands that send a specific response will be broken ?

Reply:
This will be another PR since those clusters are not included in the next test event.

Just created issue #6532 and we can implement it in another PR.

Reply:
As I said previously, this is not something I think we should do.

Can't we just do an other PR that apply similar logic to what you did for the `Test Cluster` onto the other call sites , and get this other PR to block this one ? Why do we want to break something that is working already ?

Reply:
Just to follow up on that: the plan of record all along has been to update the APIs we use for encoding APIs to be higher-level and then to implement them on top of the new encoding as needed.  This has literally been the plan for close to a year now...  It's pretty surprising that we did not in fact proceed with that plan.

Reply:
`[cluster] Implement IM specific response encoding in existing clusters` is included now.